stories/2022-03-27_Glass_ML/ML_20th_Century.ipynb

```diff
@@ -0,0 +1,2194 @@
+{
```

Comment:
<p>Jennifer Ding:</p><p>Hello! I've added some copy suggestions for tightening up and clarifying the text in the initial sections. I've added this in my comments, but I want to emphasize this one point:</p><p>It is an interesting caveat that is explicit in the title, but not really addressed in depth in the text, and I believe it's worth expanding on - typically ML trains on older data to predict on newer data, but you have turned the tables in an interesting way by training on newer (available) data to predict on an older data artefact. This is worth highlighting!</p><p> </p><p>Joe Cerniglia:</p><p>That is an interesting point. Mostly, it has to do with what was available to use as reference data. It would have been preferable, of course, to use training data contemporaneous with the artifacts, but I am not aware of any such datasets that are available. I can and should acknowledge my slightly unusual approach and highlight the fact that the chronological separation between the data creates opportunities for comparative research into the chemical elements used historically in glassmaking.</p><p> </p><p>Jennifer Ding revision:</p><p>In summer 2010, researchers from The International Group for Historic Aircraft Recovery excavated fragments of a small, semi-opaque cosmetic jar from the Pacific island of Nikumaroro.</p><p>Original:</p><p>In the summer of 2010, researchers from The International Group for Historic Aircraft Recovery excavated, in fragments, a small semi-opaque cosmetic jar from the Pacific island of Nikumaroro.</p><p> </p><p>Joe Cerniglia:</p><p>Suggested final:</p><p>In the summer of 2010, researchers from The International Group for Historic Aircraft Recovery excavated, in fragments, a small and semi-opaque cosmetic jar from the Pacific island of Nikumaroro.</p><p> </p><p>I would like to keep the phrase "in fragments" offset by commas because that is an important part of the archaeological context in which it was discovered, and I would like to highlight this. "In the summer of 2010" is slightly more ornate than "In summer 2010," but this is my first sentence. This small amount of extravagance will not be taken amiss.</p><p>====</p><p>Jennifer Ding revision:</p><p>They were searching for evidence to support their hypothesis that Amelia Earhart and Fred Noonan, her navigator, perished there in 1937 whilst on their mission to circumnavigate the globe by air along the equator, its widest point.</p><p>Original:</p><p>They were searching for evidence to support their hypothesis that Amelia Earhart and her navigator Fred Noonan perished there in 1937 after very nearly succeeding in their pathbreaking mission to circumnavigate the globe by air at its widest point, the equator.</p><p> </p><p>Joe Cerniglia:</p><p>Suggested final: I would go with the original here. The appositive "Fred Noonan" is nearly essential to the word navigator, so I would suggest no comma. The fact that Earhart very nearly succeeded is important information and I feel that should be retained.</p><p>====</p><p>Jennifer Ding revision:</p><p>My personal interest in the island's glass artefacts led me to become the team's resident glass expert, and in partnership with archaeologists Thomas King and Bill Lockhart, and chemist Greg George, we produced a&nbsp;<a href="https://www.academia.edu/40823470/A_Freckle_In_Time_or_a_Fly_in_the_Ointment" target="_blank">paper</a>&nbsp;that summarized our findings. By then, our work on the jar had already generated some&nbsp;<a href="https://www.nbcnews.com/id/wbna47623025" target="_blank">press</a>.</p><p>Original:</p><p>My interest in glass artifacts from the island ultimately led to a personal quest to become a resident glass expert of the team. In partnership with archaeologists Thomas King and Bill Lockhart, and chemist Greg George, we produced a&nbsp;<a href="https://www.academia.edu/40823470/A_Freckle_In_Time_or_a_Fly_in_the_Ointment" target="_blank">paper</a>&nbsp;that summarized our findings. By then, our work on the jar had already generated some&nbsp;<a href="https://www.nbcnews.com/id/wbna47623025" target="_blank">press</a>.</p><p>Suggested final: I would go with the original here. The revision suggests I became the team's resident glass expert, rather than "a resident glass expert." The fact is, however, that I would always grant the title of "the" resident glass expert to Bill Lockhart, an archaeologist with decades of experience and field work in glass.</p><p> </p><p>One additional note: The reason that I used Americanized spellings is that I am American. Therefore, to me "artifact" is more natural than "artefact." I realize, however, that to differently trained eyes this may look out of place. Perhaps a note could be made at the start of the paper of the fact that I hail from the States?</p><p>====</p><p>Jennifer Ding revision:</p><p>As we continued our research in 2013, I worked with&nbsp;<a href="https://www.eag.com/" target="_blank">EAG Laboratories</a>&nbsp;to determine the chemical composition of the artefact jar, and a 'sibling' glass jar that I had purchased on eBay and had named the clear facsimile. While the two jars were similar in most respects, importantly the clear facsimile jar was transparent. As EAG's&nbsp;<a href="https://tighar.org/Projects/Earhart/Archives/Research/ResearchPapers/freckleintime/Document_02_Facsimileclearjarreport.pdf" target="_blank">report</a>&nbsp;shows, while both jars were manufactured by the Hazel-Atlas company of Wheeling, West Virginia, the two jars turned out to have very different chemical compositions.</p><p>Original:</p><p>While researching the jar in 2013, I requested EAG Labs of Sunnyvale, California to determine the chemical composition of the artifact jar, and that of a 'sibling' glass jar that I had purchased on eBay and had named the clear facsimile. The clear facsimile jar was similar to the artifact in most respects, but importantly, it was transparent while the artifact was semi-opaque. As their&nbsp;<a href="https://tighar.org/Projects/Earhart/Archives/Research/ResearchPapers/freckleintime/Document_02_Facsimileclearjarreport.pdf" target="_blank">report</a>&nbsp;shows, the two jars turned out to have very different chemical compositions. Interestingly, both jars were manufactured by the same company, Hazel-Atlas of Wheeling, West Virginia.</p><p>Suggested final:</p><p>As we continued our research in 2013, I worked with&nbsp;<a href="https://www.eag.com/" target="_blank">EAG Laboratories</a>&nbsp;to determine the chemical composition of the artifact jar, and a 'sibling' glass jar that I had purchased on eBay and had named the clear facsimile. While the two jars were similar in most respects, importantly the clear facsimile jar was transparent, while the artifact was semi-opaque. As EAG's&nbsp;<a href="https://tighar.org/Projects/Earhart/Archives/Research/ResearchPapers/freckleintime/Document_02_Facsimileclearjarreport.pdf" target="_blank">report</a>&nbsp;shows, while both jars were manufactured by the Hazel-Atlas company of Wheeling, West Virginia, the two jars turned out to have very different chemical compositions.</p><p> </p><p>I generally prefer your revision, Jennifer, to my original. I merely left in the part about the artifact presenting as semi-opaque, in contrast to the sibling, and Americanizing the spelling of "artifact."</p><p>====</p><p>Jennifer Ding revision:</p><p>In the social sciences, "data" often means a collection of facts. In industry, however, we often consider data to be a collection of numbers and vectors, which we use computational tools like statistics and machine learning to analyse. As part of an eCornell machine learning course, I had the opportunity to rekindle my work on the jars and utilise the EAG labs dataset for a machine learning classification problem.</p><p>Original:</p><p>Data in the social sciences means, informally, a collection of facts. In the world of business, we work not so much with a collection of facts as a collection of data vectors, and the tools to analyze this data are not so much the tools of the laboratory as they are the tools of the computer: statistics, mathematics, programming languages, and, in recent years increasingly, machine learning.</p><p>Suggested final:</p><p>In the social sciences, "data" often mean a collection of facts. In industry, however, we often consider data to be a collection of numbers and vectors, and we use computational tools such as statistics and machine learning to analyze them. As part of an eCornell machine learning course, I had the opportunity to rekindle my work on the jars and use EAG Lab's dataset to develop a novel machine learning classification problem.</p><p> </p><p>With some minor changes, the suggested final, Jennifer, hews closely to your revision's ideas. Thanks.</p><p>====</p><p>Jennifer Ding revision:</p><p>In machine learning, classification algorithms sort data into pre-defined classes by analysing various features, or coordinate plots on a multi-dimensional plane, of the data.</p><p>Original:</p><p>One of the fundamental machine learning problems is the classification problem. Classification problems use data vectors (think of them as coordinate plots on a multi-dimensional plane), often called 'features' of the data.</p><p>Suggested final:</p><p>In this machine learning classification problem, the algorithm can assign each example of glass to a pre-defined class by analyzing relative quantities of elements from the periodic table of which each glass is made.</p><p>====</p><p>Jennifer Ding revision:</p><p>Each data point has its own unique set of features. The jars, and in this case the glass materials of the jars, offered three key measurements which represented our features: calcium, magnesium, and barium. While the EAG labs dataset was sufficient for our previous analysis, it was not well suited for a machine learning application, which typically require many examples of each class in a classification problem in order to "train" the algorithm to classify new data it has not seen before. This need for more data presented a challenge: where would I find the necessary data to effectively train a machine learning model?</p><p>Original:</p><p>Each example in the data has its own unique set of features that describe a class of the data. The jars, or more properly the glass of which the jars are made, each have three measurements for the features calcium, magnesium, and barium. For these particular instances of glass, these features may be thought to describe containers.</p><p> </p><p>Two sets of measurements for the purposes of machine learning are not of much value, however, because machine learning models require a full range of known examples upon which to 'train' a model to recognize examples it has never seen before.</p><p> </p><p>EAG Labs had been thorough, and for the purposes of our 2013 analysis, their data was ample enough. For the purposes of machine learning, however, the data, while eagerly anticipated by our team, was paltry. Where would I find the data upon which to train a machine learning model?</p><p> </p><p>Suggested final:</p><p>Each example in the data has its own unique set of features that describe a class of the data. The artifact jars, or more properly the glass of which the jars were made, each have three measurements for the features calcium, magnesium, and barium. For these particular instances of glass, these features may be thought to describe a class known as 'containers.'</p><p> </p><p>Two sets of measurements for the purposes of machine learning are not of much value, however, because machine learning models require a full range of known examples upon which to 'train' a model to recognize examples it has never seen before.</p><p> </p><p>EAG Labs had been thorough, and for the purposes of our 2013 analysis, their data was ample enough. For the purposes of machine learning, however, the data, while eagerly anticipated by our team, was paltry. Where would I find the data upon which to train a machine learning model?</p><p>====</p><p>Jennifer Ding revision:</p><p>I did not need to search very far for my answer. British forensic scientists Ian W. Evett and Ernest J. Spiehler assembled a database of glass samples in 1987 for the purposes of solving crimes, such as break-ins. Their paper was first presented at the 1987 conference of the KBS (Knowledge-Based Systems) in Goverment and is titled:</p><p>"Rule Induction in Forensic Science."&nbsp;<a href="http://gpbib.cs.ucl.ac.uk/gp-html/evett_1987_rifs.html" target="_blank">http://gpbib.cs.ucl.ac.uk/gp-html/evett_1987_rifs.html</a>. The database is available at the University of California Machine Learning Repository&nbsp;<a href="https://archive.ics.uci.edu/ml/datasets/Glass+Identification" target="_blank">here</a>. The features of their data comprised of the weight percent of eight elemental components as well as the refractive index. The types of glass represented in the sample are: Window Float, Window Non-Float, Vehicle Float, Container, Tableware, and Headlamp. There are no Vehicle Non-Float types represented in the data.</p><p>Original:</p><p>I did not need to search very far for my answer. British forensic scientists <code>Ian W. Evett</code> and<code>Ernest J. Spiehler</code> assembled a database of glass samples in 1987 for the purposes of solving crimes, such as break-ins. Their paper was first presented at the 1987 conference of the KBS (Knowledge-Based Systems) in Goverment and is titled:</p><p>"Rule Induction in Forensic Science." <a href="http://gpbib.cs.ucl.ac.uk/gp-html/evett_1987_rifs.html" target="_blank">http://gpbib.cs.ucl.ac.uk/gp-html/evett_1987_rifs.html</a>. The database is available at the University of California Machine Learning Repository <a href="http://archive.ics.uci.edu/ml/datasets/Glass+Identification" target="_blank">here</a>.</p><p>The weight percent of eight elemental components and refractive index comprise the features of their database and each sample is represented by a target variable called Type. The types of glass represented in the sample are: Window Float, Window Non-Float, Vehicle Float, Container, Tableware, and Headlamp. There are no Vehicle Non-Float types represented in the data.</p><p> </p><p>Suggested final:</p><p>These two versions are extremely close. I would like to stick with the original.</p><p>====</p><p>Jennifer Ding revision:</p><p>In comparison to this dataset, my EAG labs data was independently measured for most of the same elements as found in the U.K. database, with a few caveats, which required additional domain-specific processing in order to align the two datasets:</p><p>Original:</p><p>My two glass samples, as luck would have it, were independently measured for most of the same elements as found in the U.K. dabase [sic], with a few caveats:</p><p>Suggested final:</p><p>My two glass samples, as luck would have it, were independently measured for most of the same elements as found in the U.K. database, with a few caveats:</p><p>====</p><p>Jennifer Ding revision:</p><ul><li>Because silicon was measured in my lab data as 'Matrix,' with no actual number stated, I estimated the silicon for both containers based on the average for containers in the database (72.37).</li><li>Because K (potassium) was measured at 980 ppm for the clear facsimile, rather than by wt%, I assigned it a value of 50% of the wt% for the artifact jar: .5 X .24 = .12.</li><li>Because Fe (iron) is at very low wt% levels in the 1987 database, and the levels in my data are listed at very low parts per million (ppm), I assigned to the artifact jar and to the clear facsimile a wt% of Fe of .02 and .01, respectively.</li><li>Refractive index was not measured for either the artifact jar or the clear facsimile. Since the clear facsimile is completely transparent, I assigned it the minimum refractive index of containers from the 1987 dataset. Since the artifact jar is semi-opaque, I assigned it the maximum refractive index of containers from the 1987 dataset.</li></ul><p>These educated guesses were informed by my background in this space and were necessary to prepare the appropriate dataset for the machine learning problem demonstrated in this notebook.</p><p>Original:</p><p>Because silicon was measured in my lab data as 'Matrix,' with no actual number stated, I estimated the silicon for both containers based on the average for containers in the database (72.37). Because K (potassium) was measured at 980 ppm for the clear facsimile, rather than by wt%, I assigned it a value of 50% of the wt% for the artifact jar: .5 X .24 = .12. Because Fe (iron) is at very low wt% levels in the 1987 database, and the levels in my data are listed at very low parts per million (ppm), I assigned to the artifact jar and to the clear facsimile a wt% of Fe of .02 and .01, respectively.</p><p>Refractive index was not measured for either the artifact jar or the clear facsimile. Since the clear facsimile is completely transparent, I assigned it the minimum refractive index of containers from the 1987 dataset. Since the artifact jar is semi-opaque, I assigned it the maximum refractive index of containers from the 1987 dataset. These educated guesses were necessary for the machine learning experiment that concludes this notebook.*</p><p>Suggested final:</p><p>I agree that having bullet points is a very nice idea, and it retains the same text as the original. For that last sentence, I agree it needs elaboration. I would suggest:</p><p>These educated guessed were necessary to ensure the ability of the machine learning algorithm to compare the data from EAG Labs with the training dataset from the U.K.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/alan-turing-institute/TuringDataStories/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Reply:
Looks great, @joecerniglia !